### PR TITLE
detect-dnp3/v1: add tests

### PR DESCRIPTION
**Pull request by TheKharleeci**
Conversions:

dnp3-dnp3_data-alert: DetectDNP3DataTest01-02
dnp3-dnp3_obj-alert: DetectDNP3ObjSetupTest

### CODE REVIEW

**Comment by TheKharleeci** [comment]
Corresponding Suricata-verify PR: #OISF/suricata/pull/7057

File: tests/dnp3-dnp3_data-alert/test.yaml [hunk 1]
```diff
@@ -3,8 +3,39 @@ requires:
...
+      count: 1
+      match:
+        event_type: alert
+        alert.signature_id: 5
+        pcap_cnt: 17
+  - filter:
+      count: 0
+      match:
+        event_type: alert
+        alert.signature_id: 6
```
Comment (jufajardini) [inline tests/dnp3-dnp3_data-alert/test.yaml:36]: In `DetectDNP3DataTest02`, I see that we expect the signature that is close to our sid:6 here to match. So I think we should change that signature to have a content that would match, maybe, then add a check for the pcap_cnt, like you did for sid:7.

**Review by jufajardini — CHANGES_REQUESTED** [review]
I think this one needs a bit of work (not much!). See inline comment.

About DetectDNP3ObjSetupTest: it doesn't look to me like a test that requires conversion. (What do you say, @inashivb ? )

File: tests/dnp3-dnp3_data-alert/test.yaml (hunk 1 above)
Comment (TheKharleeci) [reply tests/dnp3-dnp3_data-alert/test.yaml:36]: Sid:6 is not expected to match based on the flow and the content. only sid:5 is expected to match

File: tests/dnp3-dnp3_obj-alert/suricata.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+%YAML 1.1
+---
+
+include: ../../etc/suricata-3.1.2.yaml
```
Comment (inashivb) [inline tests/dnp3-dnp3_obj-alert/suricata.yaml:4]: Q: Why was this needed?

File: tests/dnp3-dnp3_obj-alert/suricata.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+outputs:
+  - eve-log:
+      enabled: yes
+      filetype: regular
+      filename: eve.json
+      types:
+        - alert:
+            payload: yes
+            packet: yes
+            dnp3: yes
```
Comment (inashivb) [inline tests/dnp3-dnp3_obj-alert/suricata.yaml:15]: This looks way different than what we have in `suricata.yaml` now. Perhaps this kind of structure was followed in old revs? ref: https://github.com/OISF/suricata/blob/master/suricata.yaml.in#L156

Comment (TheKharleeci) [reply tests/dnp3-dnp3_obj-alert/suricata.yaml:15]: yes, I just followed the format that was in the previous test. Do I change it for all the dnp3 tests or just the ones I updated.

Comment (inashivb) [reply tests/dnp3-dnp3_obj-alert/suricata.yaml:15]: Perhaps, just this one for now. Thanks! :)

File: tests/dnp3-dnp3_data-alert/test.yaml (hunk 1 above)
Comment (jufajardini) [reply tests/dnp3-dnp3_data-alert/test.yaml:36]: You have it sorted out, sorry for the noise :P

**Comment by TheKharleeci** [comment]
Followed by: #766
